### PR TITLE
Basic topic controller

### PR DESCRIPTION
**Pull request by xzhflying**

### CODE REVIEW

File: app/views/course/forum/forums/_controls.html.slim [hunk 1]
```diff
@@ -3,12 +3,13 @@ span.btn-group
     = link_to unsubscribe_course_forum_path(current_course, @forum),
               title: t('course.forum.forums.unsubscribe.tag'), class: ['btn', 'btn-danger'],
               method: :delete do
-      = fa_icon 'eye-slash'.freeze
+      = fa_icon 'heart-o'.freeze
```
Comment (xzhflying) [inline app/views/course/forum/forums/_controls.html.slim]: I just change the picture of the forum subscribe button in this PR. And I'll modify these buttons in next PR following the method you provided. http://stackoverflow.com/questions/20821171/bootstrap-button-group-styling-with-button-to
@lowjoel

File: app/controllers/course/forum/forums_controller.rb [hunk 2]
```diff
@@ -10,7 +8,8 @@ def index
 
   def show
     @topics = @forum.topics.accessible_by(current_ability).order(created_at: :desc).
-              page(params[:page])
+              calculated(:post_count, :view_count).includes(:creator).page(params[:page]).
```
Comment (lowjoel) [inline app/controllers/course/forum/forums_controller.rb]: can we have a with_topic_statistics or something like that?

File: app/views/course/forum/forums/_controls.html.slim (hunk 1 above)
Comment (lowjoel) [inline app/views/course/forum/forums/_controls.html.slim]: Please change it across the entire site where possible, thanks.

Comment (xzhflying) [inline app/views/course/forum/forums/_controls.html.slim]: Ok I'll find as many as I can.

File: app/controllers/course/forum/forums_controller.rb (hunk 2 above)
Comment (xzhflying) [inline app/controllers/course/forum/forums_controller.rb]: Possible, can put `calculated(:post_count, :view_count).includes(:creator)` together.

Comment (lowjoel) [inline app/controllers/course/forum/forums_controller.rb]: no, split them.

File: config/locales/en/course/forum/topics.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+en:
+  course:
+    forum:
+      topics:
+        topic:
+          started_by: 'Started by '
+          last_post: 'Last Post: '
```
Comment (lowjoel) [inline config/locales/en/course/forum/topics.yml]: These should be parameter strings.

Comment (lowjoel) [inline config/locales/en/course/forum/topics.yml]: @xzhflying fix in the next PR.

Comment (lowjoel) [inline config/locales/en/course/forum/topics.yml]: @allenwq @xzhflying in general localisation strings should not have trailing spaces or anything like that, they should be completely standalone.

Comment (xzhflying) [inline config/locales/en/course/forum/topics.yml]: Thanks. But the problem is how to generate a link. For example, `Started by: someone`. Someone should be a link. How can I pass a link_to method as the parameter. They will be treated as strings but not html. @lowjoel

Comment (lowjoel) [inline config/locales/en/course/forum/topics.yml]: then the parameter is called something_html. See the Rails i18n guides.

Comment (xzhflying) [inline config/locales/en/course/forum/topics.yml]: OK, thanks.

**Pull request merged**